src/algorithms.ts

```diff
@@ -50,3 +51,38 @@ export function intersectTokenScores(arrays: TokenScore[][]): TokenScore[] {
 
   return result;
 }
+
+export function prioritizeTokenScores(arrays: TokenScore[][]): TokenScore[] {
+  const tokenMap: Record<string, number> = {};
+
+  const mapsLength = arrays.length;
+  for (let i = 0; i < mapsLength; i++) {
+    const arr = arrays[i];
+
+    const entriesLength = arr.length;
+    for (let j = 0; j < entriesLength; j++) {
+      const [token, score] = arr[j];
+
+      if (token in tokenMap) {
+        tokenMap[token] += score + 0.5
+      } else {
+        tokenMap[token] = score;
+      }
+    }
+  }
+
+  return Object.entries(tokenMap).sort((a, b) => b[1] - a[1]);
+}
+
+export function BM25(
+  tf: number,
+  matchingCount: number,
+  docsCount: number,
+  fieldLength: number,
+  averageFieldLength: number,
+  BM25Params: BM25Params
+) {
+  const { k, b, d } = BM25Params;
+  const idf = Math.log((docsCount - matchingCount + 0.5) / (matchingCount + 0.5));
```

Comment:
You mistakenly removed a `+ 1` from this line here: https://github.com/lucaong/minisearch/blob/1eb584c749ead8209fe4f18132e3f3a693df395e/src/MiniSearch.ts#L1854

Reply:
You're right, I didn't bring it back from here: https://github.com/LyraSearch/lyra/blob/main/src/methods/search.ts#L168

It's not covered by tests yet, and it's the reason why the PR is still open (alongside other missing features)